hathor/verification/verification_service.py

```diff
@@ -226,24 +248,31 @@ def verify_without_storage(self, vertex: BaseTransaction) -> None:
             case _:
                 assert_never(vertex.version)
 
-    def _verify_without_storage_block(self, block: Block) -> None:
-        """ Run all verifications that do not need a storage.
-        """
-        self.verifiers.vertex.verify_pow(block)
+    def _verify_without_storage_base_block(self, block: Block) -> None:
         self.verifiers.block.verify_no_inputs(block)
         self.verifiers.vertex.verify_outputs(block)
         self.verifiers.block.verify_output_token_indexes(block)
         self.verifiers.block.verify_data(block)
         self.verifiers.vertex.verify_sigops_output(block)
 
+    def _verify_without_storage_block(self, block: Block) -> None:
+        """ Run all verifications that do not need a storage.
+        """
+        self.verifiers.vertex.verify_pow(block)
+        self._verify_without_storage_base_block(block)
+
     def _verify_without_storage_merge_mined_block(self, block: MergeMinedBlock) -> None:
         self.verifiers.merge_mined_block.verify_aux_pow(block)
         self._verify_without_storage_block(block)
 
+    def _verify_without_storage_poa_block(self, block: PoaBlock) -> None:
+        self._verify_without_storage_base_block(block)
+
     def _verify_without_storage_tx(self, tx: Transaction) -> None:
         """ Run all verifications that do not need a storage.
         """
-        self.verifiers.vertex.verify_pow(tx)
+        if self._settings.CONSENSUS_ALGORITHM.is_pow():
```

Comment:
Why don't we keep the tx weight verification?

Reply:
Since there's no block mining in a PoA network, the idea was to remove transaction mining, too. To do that, we use `MIN_TX_WEIGHT=0` in settings and disable tx weight verification.

By doing this, we also remove the requirement to run a TxMiningService, simplifying use cases — at least from the full node perspective, but in practice the wallet-lib still requires a TxMiningService. This dependency can be removed from the wallet in the future so the use case would not need a TxMiningService.

Reply:
Got it. Thanks for clarifying.